### DISCUSSION TITLE
Metal render has graphical glitches on MacOs

### DISCUSSION THREAD

**Discussion by theredpro-sj** [ppy/osu#30451] [Q&A]
### Type

Game behaviour

### Bug description

Firstly I am unsure what causes the bug other than it requires the renderer to be set to metal. Sometimes when objects are moving there is some graphical glitches or screen tearing. The bug itself is most prevalent in catch the beat whenever the catcher is moving, however it does occur in other gamemodes such as in standard where it can be seen on spinners in the editor (whether not you see it is dependent on the skin your using), and in game on sliders (less noticable). I believe it might also happen in mania although I am not sure.

### Screenshots or videos

Due to file size of the videos I uploaded them using catbox
https://files.catbox.moe/ytemhp.MOV
https://files.catbox.moe/siaq68.MOV

### Version

2024.1009.1-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17561524/compressed-logs.zip)

**Comment by peppy** [comment]
What is your frame limiter and game threading mode set to? Defaults?

**Reply by theredpro-sj** [reply]
It was set to 8x refresh rate and multithreaded. I tried also on vysnc and it seems to fix the issue for catch but it sometimes still there for spinners in the std editor depending on which skin I used. Edit after trying a bit more it seems that using anything but vsync still has the issue present in catch. Using singlethreaded made no difference in the whether the bug happened or not.

**Reply by theredpro-sj** [reply]
Now that I have played a bit more it seems that the glitch does happen on sliders balls in std and is consistent on when it happens with them (ie certain sliders balls will always be glitchy while other never)

**Reply by theredpro-sj** [reply]
This also is very noticable in the mania editor timeline when using the same rendering options I have previously mentioned.
[compressed-logs.zip](https://github.com/user-attachments/files/17856649/compressed-logs.zip)


https://github.com/user-attachments/assets/2bd46800-da25-4b27-aaa4-3a342e64bc0b

**Comment by shinmorisawa** [comment]
Any other frame limiter other than VSync causes screen tearing. This is normal behavior and nothing can be done about it.

**Reply by theredpro-sj** [reply]
Just curious, but why is this the case and why does it not happen when using openGL as the renderer?

**Reply by shinmorisawa** [reply]
> Just curious, but why is this the case and why does it not happen when using openGL as the renderer?

i actually don't know, as i haven't played any games on macos in a year

**Reply by shinmorisawa** [reply]
i'm a linux + vulkan type of guy now
